apps/app_voicemail.c

```diff
@@ -3887,6 +4026,43 @@ static int msg_id_incrementor;
 static void generate_msg_id(char *dst);
 
 #ifdef ODBC_STORAGE
+
+/*!
+ * \internal
+ * \brief Create a buffer containing the SQL statement with the table name inserted.
+ *
+ * \param __sql_fmt The SQL statement with a single '%s' where the table name should be inserted.
+ *
+ * \note The buffer is allocated on the stack and should not be freed.
+ *
+ * \return A pointer to a buffer containing the SQL statement with the table name inserted.
+ */
+#define MAKE_SQL_PTRA(__sql_fmt) \
```

Comment:
Why not pass the arguments here more generically using ...?

Then you can MAKE_SQL_PTRA("UPDATE %s SET msg_id=? WHERE dir=? AND msgnum=?", odbc_table); iinstead, but the advantage is that you can nuke MAKE_SQL_PTRA2 as well, and deal with it there, even better might to turn this into an inline function with attribute's so gcc can check formats?

Reply:
See below.